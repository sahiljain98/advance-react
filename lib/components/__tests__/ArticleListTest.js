import React from 'react';
import ArticleList from '../ArticleList';
import {shallow} from 'enzyme';

describe('ArticleList', () => {
  const testProps = {
    articles: {
      a: { id: '1' },
      b: { id: '2' },
    }
  };
  it('render correctly', () => {
    const wrapper = shallow(<ArticleList {...testProps}
    />);    
    expect(wrapper.find('ArticleContainer').length).toBe(2);
    expect(wrapper).toMatchSnapshot();
  });

});
