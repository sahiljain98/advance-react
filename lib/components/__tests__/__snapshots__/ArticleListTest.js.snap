// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <ArticleContainer
        article={
            Object {
                "id": "1",
              }
        }
    />
    <ArticleContainer
        article={
            Object {
                "id": "2",
              }
        }
    />
</div>,
  "nodes": Array [
    <div>
      <ArticleContainer
            article={
                  Object {
                        "id": "1",
                      }
            }
      />
      <ArticleContainer
            article={
                  Object {
                        "id": "2",
                      }
            }
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_didScheduleRenderPhaseUpdate": false,
    "_dispatcher": Object {
      "readContext": [Function],
      "useCallback": [Function],
      "useContext": [Function],
      "useDebugValue": [Function],
      "useDeferredValue": [Function],
      "useEffect": [Function],
      "useImperativeHandle": [Function],
      "useLayoutEffect": [Function],
      "useMemo": [Function],
      "useReducer": [Function],
      "useRef": [Function],
      "useResponder": [Function],
      "useState": [Function],
      "useTransition": [Function],
    },
    "_element": <ArticleList
      articles={
            Object {
                  "a": Object {
                    "id": "1",
                  },
                  "b": Object {
                    "id": "2",
                  },
                }
      }
/>,
    "_firstWorkInProgressHook": null,
    "_forcedUpdate": false,
    "_instance": ArticleList {
      "context": Object {},
      "props": Object {
        "articles": Object {
          "a": Object {
            "id": "1",
          },
          "b": Object {
            "id": "2",
          },
        },
      },
      "refs": Object {},
      "state": null,
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_isReRender": false,
    "_newState": null,
    "_numberOfReRenders": 0,
    "_renderPhaseUpdates": null,
    "_rendered": <div>
      <ArticleContainer
            article={
                  Object {
                        "id": "1",
                      }
            }
      />
      <ArticleContainer
            article={
                  Object {
                        "id": "2",
                      }
            }
      />
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "_workInProgressHook": null,
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ArticleList
    articles={
        Object {
            "a": Object {
              "id": "1",
            },
            "b": Object {
              "id": "2",
            },
          }
    }
/>,
}
`;
